File: 项目文件/frontend/src/types/project-proposal.ts
/** 项目提案类型定义 */

/** 附件链接（地址 + 说明） */
export interface AttachmentLink {
  url: string;
  description?: string | null;
}

export interface ProjectProposal {
  id: string;
  project_id: string;
  number: string;
  title: string;
  type: string;
  priority: string;
  description: string | null;
  status: string;
  reject_reason: string | null;
  attachment_links: AttachmentLink[];
  creator_id: string;
  assignee_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectProposalCreate {
  project_id: string;
  number: string;
  title: string;
  type?: string;
  priority?: string;
  description?: string;
  status?: string;
  reject_reason?: string;
  attachment_links?: AttachmentLink[];
  assignee_id?: string;
}

export interface ProjectProposalUpdate {
  number?: string;
  title?: string;
  type?: string;
  priority?: string;
  description?: string;
  status?: string;
  reject_reason?: string;
  attachment_links?: AttachmentLink[];
  assignee_id?: string;
}

export interface ProjectProposalListResponse {
  items: ProjectProposal[];
  total: number;
}

File: 项目文件/backend/app/schemas/project_proposal.py
"""项目提案 Pydantic 模型"""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class ProjectProposalCreate(BaseModel):
    """创建项目提案请求"""

    project_id: uuid.UUID
    number: str
    title: str
    type: str = "feature"
    priority: str = "P2"
    description: str | None = None
    status: str = "pending"
    reject_reason: str | None = None
    attachment_links: list[dict[str, Any]] = []
    assignee_id: uuid.UUID | None = None


class ProjectProposalUpdate(BaseModel):
    """更新项目提案请求"""

    number: str | None = None
    title: str | None = None
    type: str | None = None
    priority: str | None = None
    description: str | None = None
    status: str | None = None
    reject_reason: str | None = None
    attachment_links: list[dict[str, Any]] | None = None
    assignee_id: uuid.UUID | None = None


class ProjectProposalResponse(BaseModel):
    """项目提案响应"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    number: str
    title: str
    type: str
    priority: str
    description: str | None = None
    status: str
    reject_reason: str | None = None
    attachment_links: list[Any] = []
    creator_id: uuid.UUID
    assignee_id: uuid.UUID | None = None
    created_at: datetime
    updated_at: datetime


class ProjectProposalListResponse(BaseModel):
    """项目提案列表响应"""

    items: list[ProjectProposalResponse]
    total: int
